src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractListItemFactory.cs

```diff
@@ -589,6 +589,26 @@ private ImmutableArray<INamedTypeSymbol> GetAccessibleTypeMembers(INamespaceOrTy
             return builder.ToImmutable();
         }
 
+        private bool IncludeTypeMember(INamedTypeSymbol typeMember, IAssemblySymbol assemblySymbol)
```

Comment:
💭 There are two ways to approach this:

* If `typeMember.Locations.Any(l => l.IsInSource)` is "faster", then the current approach is good
* If `typeMember.IsAccessibleWithin(assemblySymbol)` is "faster", then the following approach might be better:

```csharp
if (!IncludeSymbol(typeMember))
{
  return false;
}

if (typeMember.IsAccessibleWithin(assemblySymbol))
{
  return true;
}

return typeMember.Locations.Any(l => l.IsInSource);
```

:memo: Based on the form of the original code, I lean towards the second approach as the safer option.

Reply:
`IsAccessibleWithin` ends up calling [IsSymbolAccessibleCore](https://github.com/MaStr11/roslyn/blob/ClassViewShouldShowNonPublicNestedTypes/src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions_Accessibility.cs#L79). `IsSymbolAccessibleCore` is checking also cross assembly visibility. So this is heavy. `typeMember.Locations.Any(l => l.IsInSource)` is an lookup in an ImmutableArray and IsInSource is just `bool IsInSource { get { return SourceTree != null; } }`. So performance-wise `Locations.Any(l => l.IsInSource);` should be faster than `IsAccessibleWithin`.